app/controllers/api/DashboardsApiController.java

```diff
@@ -504,7 +538,17 @@ public Result addWidget(String dashboardId) {
                             widget = new StatisticalCountWidget(dashboard, query, timerange, description, field, statsFunction, streamId);
                         }
                         break;
-                    }
+                    case STACKED_CHART:
+                        Map<String, Object> requestConfig = request.config();
+                        String renderer = (String) requestConfig.get("renderer");
+                        String interpolation = (String) requestConfig.get("interpolation");
+                        String interval = (String) requestConfig.get("interval");
+                        List<Map<String, Object>> series = (List<Map<String, Object>>) requestConfig.get("series");
+
+                        if (!canReadStream(streamId)) return unauthorized();
```

Comment:
This should be checked first, before "unpacking" the configuration map. 
Also: codestyle (yadda, yadda, yadda :wink:).